lnwire/stfu.go

```diff
@@ -0,0 +1,81 @@
+package lnwire
+
+import (
+	"bytes"
+	"io"
+)
+
+// Stfu is a message that is sent to lock the channel state prior to some other
+// interactive protocol where channel updates need to be paused.
+type Stfu struct {
+	// ChanID identifies which channel needs to be frozen.
+	ChanID ChannelID
+
+	// Initiator is a byte that identifies whether we are the initiator of
+	// this process.
+	Initiator bool
+
+	// ExtraData is the set of data that was appended to this message to
+	// fill out the full maximum transport message size. These fields can
+	// be used to specify optional data such as custom TLV fields.
+	ExtraData ExtraOpaqueData
```

Comment:
Q: this is probably been asked before, atm the `stfu` only specifies `channel_id` and `initiator` fields, do we plan to add more customized data in this msg?

Reply:
@Roasbeef said we need it for forwards compatibility.

Reply:
Yeah it's worth it to add now, so if we serialize it to disk, then we never leave any optional bytes we don't understand off.